Sources/TuistGraph/Graph/GraphDependency.swift

```diff
@@ -206,11 +208,11 @@ public enum GraphDependency: Hashable, CustomStringConvertible, Comparable, Coda
         }
     }
 
-    public var isPrecompiledDynamicAndLinkable: Bool {
+    public var isPrecompiledAndEmbeddable: Bool {
         switch self {
         case .macro: return false
         case let .xcframework(xcframework):
-            return xcframework.linking == .dynamic
+            return xcframework.linking == .dynamic || xcframework.isExternal
```

Comment:
From what I can see, we need to also restrict by xcframework type framework  (`.framework`)

| XCFramework Type | Result | Note |
| ----- | --- | --- |
| Static Library `.a`| ❌ | Embedding simply copies the `.a` to `/Frameworks` with all the symbols included |
| Dynamic Library `.dylib`| ✅ | Embedding copies the `.dylib` to `/Frameworks` as expected |
| Static Framework `.framework`| ✅ | Embedding copies the `.framework` to `/Frameworks` and strips symbols from framework binary |
| Dynamic Framework `.framework`| ✅ | Embedding copies the `.framework` to `/Frameworks` as expected |

e.g. [`MyStaticLibrary.xcframework`](https://github.com/tuist/tuist/tree/main/fixtures/ios_app_with_xcframeworks/XCFrameworks/MyStaticLibrary) gets processed to a `libMyStaticLibrary.a` which shouldn't be embed. 

It doesn't look like we capture the framework type information in the xcframework metadata - we can base it based on the library extension

```suggestion
            return xcframework.linking == .dynamic || (xcframework.isExternal && xcframework.infoPlist.libraries.first?.path.extension == "framework")
```

or can uplift this to be a property on the metadata

```suggestion
            return xcframework.linking == .dynamic || (xcframework.isExternal && xcframework.type == .framework)
```

Reply:
Gotcha 👍 I updated the PR with the first suggestion

Reply:
please see also the wrong linking for xcframeworks https://github.com/tuist/tuist/pull/6153/files

Reply:
maybe xcframework.linking is just wrong